class/world.lua

```diff
@@ -77,12 +80,22 @@ World = Class {
         for i = #self.enemies, 1, -1 do
             self.enemies[i]:update(dt, self.grid:getCell(self.grid:calculateGridCoordinatesFromWorld(self.enemies[i].worldOrigin.x, self.enemies[i].worldOrigin.y)))
             if self.enemies[i].markedForDeath then
+                self.money = self.money + self.enemies[i].yield
+                table.insert(self.floatingGains, self.enemies[i]) 
```

Comment:
Inserting the existing enemy table here is not a good idea. This means keeping all of that enemy in memory until its gold counter disappeared, which is not efficient/could have some silly byproducts down the track. The use of a table to represent these is great and otherwise really well implemented so good work. (I'll make this change in my next commit)